src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -0,0 +1,288 @@
+            }
+
+            // IEnumerable<T>.GetEnumerator()
+            static BoundStatement generateGetEnumeratorT(SyntheticBoundNodeFactory f, MethodSymbol method)
+            {
+                // PROTOTYPE: Test missing member.
+                var getEnumerator = (MethodSymbol)method.DeclaringCompilation.GetSpecialTypeMember(SpecialMember.System_Collections_Generic_IEnumerable_T__GetEnumerator);
+                var field = method.ContainingType.GetFieldsToEmit().Single();
+                // return _items.GetEnumerator();
+                return f.Return(
+                    f.Call(
+                        f.Field(f.This(), field),
+                        getEnumerator));
+            }
+
+            // IReadOnlyCollection<T>.Count, ICollection<T>.Count
+            static BoundStatement generateCount(SyntheticBoundNodeFactory f, MethodSymbol method)
+            {
+                var field = method.ContainingType.GetFieldsToEmit().Single();
+                // return _items.Length;
+                return f.Return(
+                    f.ArrayLength(
+                        f.Field(f.This(), field)));
+            }
+
+            // ICollection<T>.IsReadOnly
+            static BoundStatement generateIsReadOnly(SyntheticBoundNodeFactory f, MethodSymbol method)
+            {
+                // return true;
+                return f.Return(f.Literal(true));
+            }
+
+            // IReadOnlyList<T>.this[], IList<T>.this[]
+            static BoundStatement generateIndexer(SyntheticBoundNodeFactory f, MethodSymbol method)
+            {
+                var field = method.ContainingType.GetFieldsToEmit().Single();
+                var parameter = method.Parameters[0];
+                // return _items[index];
+                return f.Return(
+                    f.ArrayAccess(
+                        f.Field(f.This(), field),
+                        f.Parameter(parameter)));
+            }
+
+            static BoundStatement generateNotSupportedException(SyntheticBoundNodeFactory f, MethodSymbol method)
+            {
+                // PROTOTYPE: Test missing type and member.
+                var constructor = (MethodSymbol)method.DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_NotSupportedException__ctor);
+                // throw new System.NotSupportedException();
+                return f.Throw(f.New(constructor));
+            }
+
+            // PROTOTYPE: Test missing interface members.
+            var membersBuilder = ArrayBuilder<Symbol>.GetInstance();
+            membersBuilder.Add(new SynthesizedFieldSymbol(this, arrayType, "_items"));
+            membersBuilder.Add(new SynthesizedReadOnlyListConstructor(this, arrayType));
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_IEnumerable__GetEnumerator), (f, m) => generateGetEnumerator(f, m)));
```

Comment:
```suggestion
            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)compilation.GetSpecialTypeMember(SpecialMember.System_Collections_IEnumerable__GetEnumerator), generateGetEnumerator));
```

applies to the rest as well.  i personally think that's clearer, and an indication htat teh local function must match this pattern.